common/src/main/java/bisq/common/app/Capabilities.java

```diff
@@ -17,55 +17,87 @@
 
 package bisq.common.app;
 
-import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 import java.util.stream.Collectors;
 
-import lombok.Getter;
-import lombok.Setter;
-
+/**
+ * hold a set of capabilities and offers appropriate comparison methods.
+ *
+ * @author Florian Reimair
+ */
 public class Capabilities {
-    // We can define here special features the client is supporting.
-    // Useful for updates to new versions where a new data type would break backwards compatibility or to
-    // limit a node to certain behaviour and roles like the seed nodes.
-    // We don't use the Enum in any serialized data, as changes in the enum would break backwards compatibility. We use the ordinal integer instead.
-    // Sequence in the enum must not be changed (append only).
-    public enum Capability {
-        TRADE_STATISTICS,
-        TRADE_STATISTICS_2,
-        ACCOUNT_AGE_WITNESS,
-        SEED_NODE,
-        DAO_FULL_NODE,
-        PROPOSAL,
-        BLIND_VOTE,
-        ACK_MSG,
-        BSQ_BLOCK
+
+    /**
+     * The global set of capabilities, i.e. the capabilities if the local app.
+     */
+    public static final Capabilities app = new Capabilities();
```

Comment:
> I would prefer a more descriptive name here. When I read Capabilities.app it was not very clear what it means.

make a suggestion

> Also does not need to be static?

yes, as it hold the capabilities of the running Bisq itself and is used in multiple places all over the code.

> Should the class extend HashSet instead of using the field capabilities - as that is basically what it holds?

see below